extensions/smallrye-fault-tolerance/deployment/src/main/java/io/quarkus/smallrye/faulttolerance/deployment/SmallRyeFaultToleranceProcessor.java

```diff
@@ -150,7 +148,10 @@ public void logCleanup(BuildProducer<LogCleanupFilterBuildItem> logCleanupFilter
 
     @Record(ExecutionTime.STATIC_INIT)
     @BuildStep
-    public void clearStatic(SmallryeFaultToleranceRecorder recorder, ShutdownContextBuildItem context) {
+    public void clearStatic(SmallryeFaultToleranceRecorder recorder, ShutdownContextBuildItem context,
+            BeanContainerBuildItem beanContainer) {
+        // impl note - we depend on BeanContainerBuildItem to make sure Arc registers before SR FT
+        // this is needed so that shutdown context of FT is executed before Arc container shuts
```

Comment:
```suggestion
        // this is needed so that shutdown context of FT is executed before Arc container shuts down
```
but that's really minor. If you could amend your existing commit, that would be perfect.

Reply:
Done!